src/Integration.Vsix/SonarLintTagger/CancellableJobRunner.cs

```diff
@@ -26,161 +26,180 @@
-        public struct JobRunnerProgress
-        {
-            public JobRunnerProgress(RunnerState currentState, int completed, int totalOperations)
-            {
-                CurrentState = currentState;
-                CompletedOperations = completed;
-                TotalOperations = totalOperations;
-            }
+        /// <summary>
+        /// Runs a series of operations on a separate background thread.
+        /// Can be cancelled.
+        /// </summary>
+        ICancellableJob Start(string jobDescription, IEnumerable<Action> operations, IProgress<JobRunnerProgress> progress, ILogger logger);
+    }
 
-            public RunnerState CurrentState { get; }
-            public int CompletedOperations { get; }
-            public int TotalOperations { get; }
-        }
+    public interface ICancellableJob
+    {
+        void Cancel();
+    }
 
-        public enum RunnerState
+    // Not overriding Equals etc - this type is only used for progress reporting.
+    // It's not intended to be stored or compared
+    public struct JobRunnerProgress
+    {
+        public JobRunnerProgress(RunnerState currentState, int completed, int totalOperations)
         {
-            NotStarted, Running, Finished, Cancelled, Faulted
+            CurrentState = currentState;
+            CompletedOperations = completed;
+            TotalOperations = totalOperations;
         }
 
-        private readonly IEnumerable<Action> operations;
-        private readonly ILogger logger;
-        private readonly string jobDescription;
-        private readonly IProgress<JobRunnerProgress> progress;
-        
-        private CancellationTokenSource cancellationSource;
-        private DateTime startTime;
+        public RunnerState CurrentState { get; }
+        public int CompletedOperations { get; }
+        public int TotalOperations { get; }
+    }
 
-        private int completedOperations;
-        private readonly int totalOperations;
+    public enum RunnerState
+    {
+        NotStarted, Running, Finished, Cancelled, Faulted
+    }
 
-        internal /* for testing */ WaitHandle TestingWaitHandle
-        { 
-            get
-            {
-                // NB this will throw if the cancellation source has
-                // been disposed but not yet nulled out
-                return cancellationSource?.Token.WaitHandle;
-            }
-        }
-        internal /* for testing */ RunnerState State { get; private set; }
+    internal sealed class CancellableJobRunner : ICancellableJobRunner
+    {
+        public static ICancellableJobRunner Instance { get; } = new CancellableJobRunner();
 
-        public static CancellableJobRunner Start(string jobDescription, IEnumerable<Action> operations, IProgress<JobRunnerProgress> progress, ILogger logger)
+        public ICancellableJob Start(string jobDescription, IEnumerable<Action> operations, IProgress<JobRunnerProgress> progress, ILogger logger)
```

Comment:
Method is now non-static, and it returns the new `ICancellableJob` - the caller doesn't need to know anything about the returned type except that it can call `Cancel` on it.